test/e2e/snaps/test-snap-txinsights-v2.spec.js

```diff
@@ -99,6 +99,12 @@ describe('Test Snap TxInsights-v2', function () {
           WINDOW_TITLES.Dialog,
           windowHandles,
         );
+
+        await driver.findClickableElement({
+          text: 'Confirm',
+          tag: 'button',
+        });
```

Comment:
could you help me understand why this change was required?

Reply:
It's all rooted in the issue of not having correct gas estimate values on first render from the hook coupled with gas estimation logic being mixed into the ui components themselves. If the gas estimation UI (the details tab) isn't mounted when the gas estimates become available, then the gas estimates don't get updated. I can't think of a good way to deal with this issue as gas estimation shouldn't be in the ui layer in the first place. @matthewwalsh0 has mentioned tackling this by moving gas estimation into the background at the TransactionController layer

Reply:
🤔 ok this is a much bigger topic than the initial question here but I think ultimately we **do** want the polling to be tied to UI components. See this [thread](https://consensys.slack.com/archives/C06KYQG05A4/p1708980778877819)

Reply:
> Tied to meaning started and stopped based on UI component mounting, yes I agree. Tied to meaning the actual gas estimates costs for a transactions being calculated by tightly coupled react hooks, no I disagree.

🤝 we are in accord.